lom/src/lib/clib/lom_clib.go

```diff
@@ -0,0 +1,254 @@
+package main
+
+import (
+    "C"
+    "encoding/json"
+    "fmt"
+    . "lom/src/lib/lomcommon"
+    . "lom/src/lib/lomipc"
+    "lom/src/engine"
+)
+
+
+/*
+ * C-bindings for non-go clients.
+ *
+ * A simple wrapper to corresponding APIs
+ */
+
+func TestCall() {
+}
+
+var LoMInitialized = 0
+var LoMRunMode = 0
+
+/*
+ * ----------------------------------------------------------------
+ * State get APIs
+ * ----------------------------------------------------------------
+ */
+//export GetLoMRunModeC
+func GetLoMRunModeC() C.int {
+    return C.int(LoMRunMode)
+}
+
+/*
+ * ----------------------------------------------------------------
+ * Config get APIs
+ * ----------------------------------------------------------------
+ */
+
+//export InitConfigPathForC
+func InitConfigPathForC(pathPtr *C.char) C.int {
+    path := C.GoString(pathPtr)
+    if err := InitConfigPath(path); err != nil {
+        LogError("Failed to init config for path(%s) err(%v)", path, err)
+        return C.int(-1)
+    }
+    LoMRunMode = int(GetLoMRunMode())
+    LoMInitialized = 1
+    return C.int(0)
+}
+
+//export GetGlobalCfgStrC
+func GetGlobalCfgStrC(keyPtr *C.char) *C.char {
+    key := C.GoString(keyPtr)
+    ret := fmt.Sprintf("%v", GetConfigMgr().GetGlobalCfgAny(key))
+    return C.CString(ret)
+}
+
+//export GetGlobalCfgIntC
+func GetGlobalCfgIntC(keyPtr *C.char) C.int {
+    key := C.GoString(keyPtr)
+    val := GetConfigMgr().GetGlobalCfgAny(key)
+    iVal, ok := val.(float64) 
+    if !ok {
+        LogError("Failed key=%s expect float64 val (%T)/(%v)", key, val, val)
+    }
+    return C.int(iVal)     /* Defaults to 0 on failed conversion */
+}
+
+//export GetSequenceAsJsonC
+func GetSequenceAsJsonC(namePtr *C.char) *C.char {
+    name:= C.GoString(namePtr)
+    ret := ""
+    if v, err := GetConfigMgr().GetSequence(name); err != nil {
+        LogError("Failed to find sequence for (%s) (%v)", name, err)
+    } else if out, err := json.Marshal(v); err != nil {
+        LogError("Failed to marshal sequence (%v) err(%v)", v, err)
+    } else {
+        ret = string(out)
+    }
+    return C.CString(ret)
+}
+
+//export GetActionConfigAsJsonC
+func GetActionConfigAsJsonC(namePtr *C.char) *C.char {
+    name:= C.GoString(namePtr)
+    ret := ""
+    if v, err := GetConfigMgr().GetActionConfig(name); err != nil {
+        LogError("Failed to get conf for action (%s) (%v)", name, err)
```

Comment:
why we need to print log here ? Since this is binding API, error can be passed to caller and it can be handled there ?

Reply:
I want to ensure it is logged.
Client expects only empty or valid JSON string.